src/storage/src/client.rs

```diff
@@ -332,10 +332,75 @@ mod v1 {
         time_storage_class_updated: wkt::Timestamp,
         updated: wkt::Timestamp,
         custom_time: wkt::Timestamp,
+        acl: Vec<ObjectAccessControl>,
         // The following are excluded from the protos, so we don't really need to parse them.
         media_link: String,
         self_link: String,
-        // TODO(#2039) - add all the other fields.
+        // TODO(#2039) - add all the other fields:
+        //     "md5Hash": string,
+        //     "crc32c": string,
+        //     "retention": {
+        //       "retainUntilTime": "datetime",
+        //       "mode": string
+        //     }
+        //     "metadata": {
+        //       (key): string
+        //     },
+        //     "owner": {
+        //       "entity": string,
+        //       "entityId": string
+        //     },
+        //     "customerEncryption": {
+        //       "encryptionAlgorithm": string,
+        //       "keySha256": string
+        //     }
+    }
+
+    #[serde_with::serde_as]
+    #[derive(Debug, Default, serde::Deserialize, PartialEq)]
+    #[serde(default, rename_all = "camelCase")]
+    struct ObjectAccessControl {
+        id: String,
+        entity: String,
+        role: String,
+        email: String,
+        domain: String,
+        entity_id: String,
+        etag: String,
+        project_team: Option<ProjectTeam>,
+    }
+
+    #[serde_with::serde_as]
+    #[derive(Debug, Default, serde::Deserialize, PartialEq)]
+    #[serde(default, rename_all = "camelCase")]
+    struct ProjectTeam {
+        project_number: String,
+        team: String,
+    }
+
+    impl std::convert::From<ObjectAccessControl> for control::model::ObjectAccessControl {
+        fn from(value: ObjectAccessControl) -> Self {
+            Self::new()
+                .set_id(value.id)
+                .set_entity(value.entity)
+                .set_role(value.role)
+                .set_email(value.email)
+                .set_domain(value.domain)
+                .set_entity_id(value.entity_id)
+                .set_etag(value.etag)
+                .set_project_team(to_control_project_team(value.project_team))
+        }
+    }
+
+    fn to_control_project_team(p: Option<ProjectTeam>) -> Option<control::model::ProjectTeam> {
```

Comment:
Was unable to do fromFrom<Option> for Option<> since this crate does not own the Option type I believe, but if there is something I was missing in order to make that work please let me know.

Reply:
Done. thank you!